lib/datadog/tracing/contrib/http/configuration/settings.rb

```diff
@@ -33,6 +33,11 @@ class Settings < Contrib::Configuration::Settings
               o.lazy
             end
 
+            option :response_code_errors do |o|
+              o.default { env_to_list(Ext::ENV_RESPONSE_CODE_ERRORS, 400...599, comma_separated_only: false)}
```

Comment:
```ruby
(400...599).include?(599)
# => false
```

The endless range does not include `599`. I suggest changing the end value to 600, keeping the range endless:

```suggestion
              o.default { env_to_list(Ext::ENV_RESPONSE_CODE_ERRORS, 400...600, comma_separated_only: false)}
```

Reply:
That makes sense 👍 

FWIW This default value of `400...599` was taken from the current behavior in order to keep this change backwards-compatible: https://github.com/DataDog/dd-trace-rb/blob/e5bb0eb9892ac9474f0887d703052eb55d3b1e6b/lib/datadog/tracing/contrib/http/instrumentation.rb#L98-L101

Do you think it's worth making a backwards-incompatible change to update this range?

Reply:
Given the exclusion of 599 is likely an off-by-one bug, it should be fixed.
Bug fixes are allowed to perform behaviour changes.